contracts/dapps/astar-dapp-staking/AstarDappStakingFactory.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.0;
+pragma experimental ABIEncoderV2;
+
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../../interfaces/IPoolFactory.sol";
+import "./AstarDappStaking.sol";
+
+/**
+ * @dev Factory contract of Astar Dapp staking pool.
+ *x
+ */
+contract AstarDappStakingFactory is IPoolFactory, Ownable {
+    using BytesLib for bytes;
+
+    address astarDAppStakingContract;
+
+    event StakingContractChange(address indexed oldContract, address indexed newContract);
+    event AStarDappStakingCreated(
+        address indexed pool,
+        address indexed community,
+        string name,
+        address dapp
+    );
+
+    constructor(address _astarDAppStakingContract) {
+        require(_astarDAppStakingContract != address(0), 'Invalid argument');
+        astarDAppStakingContract = _astarDAppStakingContract;
+    }
+
+    function adminSetStakingContract(address _astarDAppStakingContract) external onlyOwner {
+        require(_astarDAppStakingContract != address(0), 'Invalid argument');
+        address old = astarDAppStakingContract;
+        astarDAppStakingContract = _astarDAppStakingContract;
+        emit StakingContractChange(old, _astarDAppStakingContract);
+    }
+
+    function createPool(address community, string memory name, bytes calldata meta) override external returns(address) {
+        require(community == msg.sender, 'Permission denied: caller is not community');
```

Comment:
require(CommunityFactory(communityFactory).createdCommunity(community), "Invalid community");
Need to add this require, other wise any one can call this method use a invalid community id, and the event will be catch by the graph, and our data will be dirty.

Reply:
Done